pxr/base/tf/stringUtils.cpp

```diff
@@ -806,8 +806,8 @@ TfDictionaryLessThan::_LessImpl(const string& lstr, const string& rstr) const
             // Add 5 mod 32 makes '_' sort before all letters.
             return ((l + 5) & 31) < ((r + 5) & 31);
         }
-        else if (IsDigit(l) | IsDigit(r)) {
-            if (IsDigit(l) & IsDigit(r)) {
+        else if (IsDigit(l) || IsDigit(r)) {
+            if (IsDigit(l) && IsDigit(r)) {
```

Comment:
I can understand that - the change would be to disable the warning then and comment the reason for the it.

Reply:
Diving into it a bit more to explain, I was quick to go to a documentation approach. Runtime performance would be dependent on the probability that the quick outs are hit or not. There are more instructions since the logical operators must include the test and jump when compared to the bitwise. This is a case where a pure instruction equivalency would not provide the right data, as it is more of an empirical test with real-world data that would provide insights into performance in normal situations. I'm going to assume something like that was done, so documenting it sounds like the right course of action. The other factor would be non-x86 hardware and how the CPU handles branch prediction and out-of-order processing.

Reply:
For comparison and FYI:

The boolean operator (2 compare and jumps - hard to say what that looks like on x86 out-of-order performance)
        cmp     dword ptr [rbp - 8], 0
        jne     .LBB1_2
        cmp     dword ptr [rbp - 12], 0
        je      .LBB1_3

The bitwise operator (1 compare and jump, and a mov and or - same number of memory references)
        mov     eax, dword ptr [rbp - 8]
        or      eax, dword ptr [rbp - 12]
        cmp     eax, 0
        je      .LBB1_2

code:
    int varA=0, varB=1;
    if (varA | varB)